File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/departamento.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class departamento
{
    [Key]
    public int id_departamento { get; set; }

    [Required(ErrorMessage = "El nombre del departamento es obligatorio.")]
    [StringLength(255, ErrorMessage = "El nombre del departamento no puede exceder los 255 caracteres.")]
    public string nombre_departamento { get; set; } = null!;

    [Required(ErrorMessage = "El país es obligatorio.")]
    public int id_pais { get; set; }

    [InverseProperty("id_departamentoNavigation")]
    public virtual ICollection<ciudade> ciudades { get; set; } = new List<ciudade>();

    [ForeignKey("id_pais")]
    [InverseProperty("departamentos")]
    public virtual paise id_paisNavigation { get; set; } = null!;
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/departamento.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const departamentoSchema = new Schema({
    
    nombre_departamento: {
        type: String,
        required: [true, 'El nombre del departamento es obligatorio.'],
        maxlength: [255, 'El nombre del departamento no puede exceder los 255 caracteres.']
    },
    id_pais: {
        type: Schema.Types.ObjectId,
        ref: 'Paise', // Asegúrate de que este sea el nombre correcto del modelo pais
        required: [true, 'El país es obligatorio.']
    },
    ciudades: [{
        type: Schema.Types.ObjectId,
        ref: 'Ciudade' // Asegúrate de que este sea el nombre correcto del modelo ciudad
    }]
});

// Exportar el modelo
module.exports = mongoose.model('Departamento', departamentoSchema);
